### PR TITLE
Added more tests, labels and cleaned the code a bit

### PR DESCRIPTION
**Pull request by jonasbn**
# Description

I have added more tests to increase coverage

References #36 

- [X] This change is maintenance to infrastructure, documentation or test suite

# Checklist:

- [X] My code follows the style guidelines of this project, please see the [contribution guidelines](https://github.com/jonasbn/perl-workflow/blob/master/CONTRIBUTING.md).
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: t/validator_matches_date_format.t [hunk 1]
```diff
@@ -2,35 +2,46 @@
...
 
 my $validator;
 my $wf;
 
-dies_ok { $validator = Workflow::Validator::MatchesDateFormat->new({}) };
+dies_ok { $validator = Workflow::Validator::MatchesDateFormat->new({}) } 'Constructor without parameters, we die';
+
+dies_ok { $validator = Workflow::Validator::MatchesDateFormat->new({ date_format => bless {} }) } 'Constructor with bad parameters, we die';
 
 ok($validator = Workflow::Validator::MatchesDateFormat->new({
```
Comment (ehuelsmann) [inline t/validator_matches_date_format.t:18]: Isn't this a 'lives'?

**Review by ehuelsmann — APPROVED** [review]
One comment. Otherwise looking good.

File: t/validator_matches_date_format.t (hunk 1 above)
Comment (jonasbn) [reply t/validator_matches_date_format.t:18]: Hmmm it throws a configuration exception if the `date_format` parameter is not supplied to the constructor in Workflow::Validator::MatchesDateFormat

REF: https://github.com/jonasbn/perl-workflow/blob/2857db36d3dcb9204e23dfddbd22855a695d52fa/lib/Workflow/Validator/MatchesDateFormat.pm#L17-L20

**Comment by jonasbn** [comment]
Hi @ehuelsmann 

Could you please re-review after changes based on your suggestions on Slack

File: t/validator_matches_date_format.t [hunk 2]
```diff
@@ -15,13 +15,19 @@ dies_ok { $validator = Workflow::Validator::MatchesDateFormat->new({}) } 'Constr
 
 dies_ok { $validator = Workflow::Validator::MatchesDateFormat->new({ date_format => bless {} }) } 'Constructor with bad parameters, we die';
 
-ok($validator = Workflow::Validator::MatchesDateFormat->new({
+lives_and { $validator = Workflow::Validator::MatchesDateFormat->new({
```
Comment (ehuelsmann) [inline t/validator_matches_date_format.t:18]: Since there is no inner test, i think this should be `lives_ok`

**Review by ehuelsmann — CHANGES_REQUESTED** [review]
When the lives_and becomes lives_ok, this is good to go.

**Review by ehuelsmann — APPROVED** [review]

**Pull request merged**